src/utils/self_test/src/configini.c

```diff
@@ -0,0 +1,1353 @@
+/*
+ *	libconfigini - an ini formatted configuration parser library
+ *	Copyright (C) 2013-present Taner YILMAZ <taner44@gmail.com>
+
+ *	Redistribution and use in source and binary forms, with or without
+ *	modification, are permitted provided that the following conditions
+ *	are met:
+ *	 1. Redistributions of source code must retain the above copyright
+ *		notice, this list of conditions and the following disclaimer.
+ *	 2. Redistributions in binary form must reproduce the above copyright
+ *		notice, this list of conditions and the following disclaimer
+ *		in the documentation and/or other materials provided with the
+ *		distribution.
+ *	 3. Neither the name of copyright holders nor the names of its
+ *		contributors may be used to endorse or promote products derived
+ *		from this software without specific prior written permission.
+ *	THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *	"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *	LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *	A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL COPYRIGHT
+ *	HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *	SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *	LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *	DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *	THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *	(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *	OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+
+#include "configini.h"
+#include "queue.h"
+
+#define COMMENT_CHARS	"#"    /* default comment chars */
```

Comment:
We could, for the time being I think if we just keep to 1 type of comments is best.  Since this is new, just allow for one type.  No need to worry about back compatibility.